public/source/server.js

```diff
@@ -97,14 +97,47 @@ Server.prototype._getCredentials = function(callback) {
 Server.prototype.watchRepository = function(repositoryPath, callback) {
   this.socket.emit('watch', { path: repositoryPath }, callback);
 };
-Server.prototype.queryPromise = function(type, url, arg) {
+Server.prototype.queryPromise = function(method, path, body) {
   var self = this;
+  if (body) body.socketId = this.socketId;
+  var request = {
+    method: method,
+    url: rootPath + '/api' + path,
+  }
+  if (method == 'GET' || method == 'DELETE') request.query = body;
+  else request.body = body;
+
   return new Promise(function (resolve, reject) {
-    self.query(type, url, arg, function(err, result) {
-      if (err) {
-        reject(err);
+    self._httpJsonRequest(request, function(error, res) {
```

Comment:
Right, for this round I have focussed on promisifying `server.get`.  After this I will be promisifying `server.post`, `server.delete`, etc and all the way down to those methods.

I should have been more clear that this is not ready yet...  sorry.